packages/catalog-search/src/SearchBox.jsx

```diff
@@ -54,40 +59,61 @@ export const SearchBoxBase = ({
     }
   };
 
+  const handleSuggestionSubmit = (query, hit) => {
+    // If an override is provided, call it with the hit
+    if (suggestionSubmitOverride !== undefined) {
+      suggestionSubmitOverride(hit);
+    }
+
+    // When a suggested search is clicked, set the course tile to the search query
+    setSearchQuery(query);
+    // Do all the regular submit stuff
+    handleSubmit(query);
+  };
+
   useEffect(() => {
     document.addEventListener('click', () => setShowSuggestions(false));
   });
```

Comment:
This should likely have a clean up function to remove the event listener when the component is unmounted.

https://blog.logrocket.com/understanding-react-useeffect-cleanup-function/